src/components/products/Products.jsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from 'react';
+import data from '../../products.json';
+import { Container } from './styledProducts';
+import Sorting from '../sorting/Sorting';
+import Pagination from '../pagination/Pagination';
+import ProductsList from '../productsList/ProductsList';
+
+const Products = () => {
+  const [products, setProducts] = useState(data);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [productsPerPage] = useState(9);
+  const [sorting, setSorting] = useState('Default');
+
+  const lastProductIndex = currentPage * productsPerPage;
+  const firstProductIndex = lastProductIndex - productsPerPage;
+  const currentProducts = products.slice(firstProductIndex, lastProductIndex);
+
+  const paginate = (pageNumber) => setCurrentPage(pageNumber);
+
+  const prevPage = () =>
+    setCurrentPage((pageNumber) => (pageNumber === 1 ? 1 : pageNumber - 1));
+
+  const nextPage = () =>
+    setCurrentPage((pageNumber) =>
+      pageNumber === Math.ceil(products.length / productsPerPage)
+        ? pageNumber
+        : pageNumber + 1
+    );
+
+  const sortProducts = (sorting) => {
+    if (sorting === 'Default') return data;
+    const sortedProducts = [...data];
+    if (sorting === 'Price↑' || sorting === 'Price↓') {
+      sortedProducts.sort((a, b) => a.price - b.price);
+      return sorting === 'Price↑' ? sortedProducts : sortedProducts.reverse();
+    }
+
+    if (sorting === 'Name↑' || sorting === 'Name↓') {
+      sortedProducts.sort((a, b) => {
+        if (a.name > b.name) {
```

Comment:
функция сортировки

Reply:
Переделал функцию сортировки по имени:
```suggestion
        sortedProducts.sort((a, b) => a.name.localeCompare(b.name));
```